### ISSUE TITLE
Port to LLVM

### ISSUE THREAD

**Issue by hygoni** [hygoni/precise-leak-sanitizer#18]
Issue #15 should be completed before working on this issue.

# Port LLVM Pass to LLVM

**TL;DR: Make it possible to invoke precise leak sanitizer pass without loading plugin dynamically.**
**i.e. $ ./build/bin/opt --passes=precise-leak-sanitizer should work**

Technically, porting the pass (not compiler-rt) is very simple.

- [ ] Update llvm-project/llvm/lib/Transforms/Instrumentation/CMakeLists.txt
- [ ] Move files to llvm/lib/Transforms/Instrumentation/PreciseLeakSanitizer.cpp and llvm/include/llvm/Transforms/Instrumentation/PreciseLeakSanitizer.h

See [static plugin](https://github.com/banach-space/llvm-tutor?tab=readme-ov-file#dynamic-vs-static-plugins) for an example.

# Port runtime library to LLVM

Similar to #18, but this issue ports runtime library, not LLVM pass.  

- [ ] Figure out how to add new runtime library in llvm-project/compiler-rt
- [ ] Move our runtime library into compiler-rt
- [ ] Figure out how to link our runtime library (prolly using clang frontend) into the executable
